File: flink/flink_kafka_java_demo_1/src/main/java/demo/flink/config/AppConfig.java
package demo.flink.config;

import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;

public class AppConfig {

    private static Config defaultConf = ConfigFactory.load();
    private static Config envConf = ConfigFactory.systemEnvironment();
    public static Config config = envConf.withFallback(defaultConf);

    public static String getString(String key, String defValue) {
        return config.hasPath(key) ? config.getString(key) : defValue;
    }



}

File: flink/flink_kafka_scala_demo_1/src/main/scala/demo/flink/scala/one/config/AppConfig.scala
package demo.flink.scala.one.config

import com.typesafe.config.Config
import org.apache.flink.api.common.typeinfo.TypeInformation
import org.apache.flink.api.java.typeutils.TypeExtractor
import org.apache.flink.streaming.api.scala.OutputTag

import java.io.Serializable

class AppConfig(val config: Config) extends Serializable {

  implicit val stringTypeInfo: TypeInformation[String] = TypeExtractor.getForClass(classOf[String])

  def getConfig(): Config = config;
  def getString(key: String, defValue: String): String = if(config.hasPath(key)) config.getString(key) else defValue
  def getBoolean(key: String, defValue: Boolean): Boolean = if(config.hasPath(key)) config.getBoolean(key) else defValue

  val messageTransmitterTag: OutputTag[String] = OutputTag[String]("message-transmitter")
}
